### PR TITLE
Adding the possibility to redact Sitter info about themselves.

### LINKED ISSUE #44: Adjust `SitterProfileView` and `SitterProfileViewModel` to be able to change the info more than once
## This task is a part of the issue:
 - #38 

## Depends on tasks:
- #8 

### User story:
When sitters switche to `Profile` tab  they should be able to change any info about themselves on their profile page.
The button `Edit` should be visible at all times.

### Dev requirements and restrictions:
Refactor the `SitterProfileViewModel`: we need to satisfy the restriction when should make a post or a put request.
When the user changes the info the first time we should use `WoofAppEndpoint.addNewSitter`
When the user edits previously entered info we should use `WoofAppEndpoint.updateSitter`

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #44

## Changes in:
- SitterProfileViewModel, added private method `update()` to use with PUT request
- SitterProfileView, the `edit` button is always shown.

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Nice attempt 👍 
Can we add more tests, as we use `isSitterSet` property to use proper request, I think, it can be useful to check that this property will be  changed only once:
for example,
1) we can verify that the property `isSitterSet` was changed when we call `save` method only once .
2) when we call it twice, `isSitterSet` doesn't change.

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 ### Added 
 
 - Add new templates for `epic` issue and `feature request` - [4](https://github.com/IronFoundation/iOS-Woof/pull/4)
+- Add the possibility to redact `Sitter` info about themselves - [52](https://github.com/IronFoundation/iOS-Woof/pull/52)
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply CHANGELOG.md]: Changed with 1bb9960

File: Woof/Woof/ViewModel/SitterProfileViewModel.swift [hunk 2]
```diff
@@ -109,6 +113,16 @@ final class SitterProfileViewModel: ObservableObject {
         }
     }
 
+    private func update(_ sitter: Sitter) async throws {
+        let endpoint = WoofAppEndpoint.updateSitter(sitter.asDictionary())
+
+        do {
+            _ = try await NetworkService<WoofAppEndpoint>().request(endpoint)
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/ViewModel/SitterProfileViewModel.swift:120]: as we use an instance of `NetworkService` twice does it make sense to create a private property for it
` private let service = NetworkService<WoofAppEndpoint>()`

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 3]
```diff
@@ -127,4 +129,52 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
...
+        }
+
+        // When
+        let sitterIsSetBeforeSave = viewModel.sitterIsSet
+        await viewModel.save()
+        let sitterIsSetAfterSave = viewModel.sitterIsSet
+
+        // Then
+        XCTAssertFalse(sitterIsSetBeforeSave)
+        XCTAssertTrue(sitterIsSetAfterSave)
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:154]: Do we really need to create property `sitterIsSetBeforeSave` and `sitterIsSetAfterSave`?
```suggestion
         XCTAssertFalse(viewModel.sitterIsSet)

        // When
        await viewModel.save()

        // Then
        XCTAssertTrue(viewModel.sitterIsSet)
```

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 4]
```diff
@@ -127,4 +129,52 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
...
+    }
+
+    func testSecondSaveDoesntChangeSitterIsSet() async {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:157]: nit: The test name is not clear to me.
```suggestion

    func testSitterIsSetIsNotChangedWhenSaveCalledTwice() async {
```

Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift]: Changed with 7f84b49

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 3 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift]: Changed with 7f84b49

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 5]
```diff
@@ -6,11 +6,13 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
 
     override func setUp() {
         super.setUp()
+        KeyValueStorage.clearStoredSitterData()
         URLProtocol.registerClass(MockURLProtocol.self)
         viewModel = SitterProfileViewModel()
     }
 
     override func tearDown() {
+        KeyValueStorage.clearStoredSitterData()
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:15]: Why do we need to call this function in both `setUp()` and `tearDown()`?

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 6]
```diff
@@ -127,4 +129,48 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
...
+
+        // When
+        let sitterIsSetBeforeSave = viewModel.sitterIsSet
+        await viewModel.save()
+
+        // Then
+        XCTAssertNotEqual(sitterIsSetBeforeSave, viewModel.sitterIsSet)
+    }
+
+    func testSitterIsSetIsNotChangedWhenSaveCalledTwice() async {
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:155]: You don't compare the initial value and the result value of the `sitterIsSet`, thus we can't say that "`sitterIsSet` is not changed". 
```suggestion
    func testSitterIsSetIsTrueWhenSaveIsCalledRepeatedly() async {
```

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 7]
```diff
@@ -6,11 +6,13 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
 
     override func setUp() {
         super.setUp()
+        KeyValueStorage.clearStoredSitterData()
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:9]: We don't have test cases when sitter is loaded from storage, and we don't imply such possibility. Am I right and is it expected result?

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 8]
```diff
@@ -127,4 +129,48 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
         // Then
         XCTAssertEqual(requestCount, 1)
     }
+
+    func testFirstSaveChangesSitterIsSet() async {
```
Comment (kirpichonok) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:133]: nit:
```suggestion
    func testFirstSavedSitterChangesSitterIsSetPropertyOnTrue() async {
```

**Review by kirpichonok — CHANGES_REQUESTED** [review]
Good job, but answer some questions and consider improvements.

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 7 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:9]: We clear the storage in `tearDown()`, later in another test I check that the `sitterIsSet` changes when `save()` is called

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 5 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:15]: it's not obligatory, deleted `KeyValueStorage.clearStoredSitterData()` for setUp() with 34c6879

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 6 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:155]: Changed with 34c6879

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 8 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:133]: Changed with 34c6879

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/Helper/KeyValueStorageClearStoredSitterData.swift [hunk 9]
```diff
@@ -0,0 +1,7 @@
+extension KeyValueStorage {
+    /// Clears stored sitters data from storage.
+    static func clearStoredSitterData() {
+        let keyValueStorage = KeyValueStorage(KeyValueStorage.Name.currentSitter)
+        keyValueStorage.deleteData(for: KeyValueStorage.Key.currentSitter)
+    }
+}
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/Helper/KeyValueStorageClearStoredSitterData.swift]: I chose option 2 and implemented it in 97afb4b

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 3 above)
Comment (NadzeyaShpakouskaya) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift]: The current version of your test is not related to its name.
You check that properties vary, please take a look at my suggestion.

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift [hunk 10]
```diff
@@ -130,7 +129,7 @@ final class SitterProfileViewModelSaveTests: XCTestCase {
         XCTAssertEqual(requestCount, 1)
     }
 
-    func testFirstSaveChangesSitterIsSet() async {
+    func testFirstSavedSitterChangesSitterIsSetPropertyOnTrue() async {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift:132]: typo:
```suggestion
    func testFirstSavedSitterChangesSitterIsSetPropertyToTrue() async {
```

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 3 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift]: Sorry i forget about it, fixed it with 97afb4b

**Review by NadzeyaShpakouskaya — COMMENTED** [review]
Overall LGTM, after resolving all conversation

File: Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift (hunk 10 above)
Comment (iBYarchitect) [reply Woof/WoofTests/ViewModel/SitterProfileViewModelTests/SitterProfileViewModelSaveTests.swift]: Fixed with 067615c

**Review by kirpichonok — APPROVED** [review]
LGTM

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM

**Pull request merged**